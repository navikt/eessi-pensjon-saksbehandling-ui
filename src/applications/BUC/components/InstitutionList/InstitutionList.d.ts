export interface InstitutionListProps {
  className?: string;
  flag?: boolean;
  flagType?: string;
  institutions: Array<any>;
  institutionNames: InstitutionNames;
  locale: string;
  t: (...args: any[]) => any;
  type: string;
}
