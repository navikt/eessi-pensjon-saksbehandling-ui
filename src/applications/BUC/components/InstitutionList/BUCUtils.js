import { CountryData } from 'eessi-pensjon-ui'

export const getBucTypeLabel = ({ type, locale, t }) => {
  if (!type.match('P3000_')) {
    return t('buc:buc-' + type)
  }
  const country = type.match(/_(.*)$/)[1]
  console.log(type, country)
  const countryLabel = CountryData.findByValue(locale, country)
  return t('buc:buc-P3000_XX', { country: countryLabel.label })
}
