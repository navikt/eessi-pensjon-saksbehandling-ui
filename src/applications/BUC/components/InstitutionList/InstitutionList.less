.a-buc-c-institutionlist {
  display: flex;
  flex-direction: column;
  align-items: baseline;
}

.a-buc-c-institution {
  display: flex;
  flex-direction: row;
  align-items: center;
}
