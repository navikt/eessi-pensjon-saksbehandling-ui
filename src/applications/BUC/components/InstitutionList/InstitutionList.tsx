import { InstitutionListMap, InstitutionNames, Institutions } from 'declarations/buc'
import { InstitutionsPropType } from 'declarations/buc.pt'
import { State } from 'declarations/reducers'
import { AllowedLocaleString } from 'declarations/types'
import { AllowedLocaleStringPropType } from 'declarations/types.pt'
import Flag from 'flagg-ikoner'
import CountryData from 'land-verktoy'
import _ from 'lodash'
import { Normaltekst } from 'nav-frontend-typografi'
import PT from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { useSelector } from 'react-redux'
import styled from 'styled-components'

export interface InstitutionListProps {
  className?: string
  flag?: boolean
  flagType?: 'original' | 'circle'
  institutions: Institutions
  locale: AllowedLocaleString
  type?: string
}

const InstitutionListDiv = styled.div`
  display: flex;
  flex-direction: column;
`
const InstitutionDiv = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`
const InstitutionText = styled(Normaltekst)`
  margin-left: 0.5rem;
`
const InstitutionList: React.FC<InstitutionListProps> = ({
  className, flag = true, flagType = 'circle', institutions = [], locale, type = 'joined'
}: InstitutionListProps): JSX.Element => {
  const institutionList: InstitutionListMap<string> = {}
  const institutionNames: InstitutionNames = useSelector<State, InstitutionNames>(state => state.buc.institutionNames)
  const { t } = useTranslation()

  if (institutions) {
    institutions.forEach(item => {
      let institution: string = item.institution
      const country: string = item.country
      if (institution && institution.startsWith(country + ':')) {
        institution = institution.substring(institution.indexOf(':') + 1, institution.length)
      }
      if (Object.prototype.hasOwnProperty.call(institutionList, country)) {
        institutionList[country].push(institution)
      } else {
        institutionList[country] = [institution]
      }
    })
  }

  const getLabel = (landkode: string, institutionId: string): string => {
    return institutionNames &&
    Object.prototype.hasOwnProperty.call(institutionNames, landkode + ':' + institutionId)
      ? institutionNames[landkode + ':' + institutionId]
      : institutionId
  }

  return _.isEmpty(institutionList) ? (
    <InstitutionListDiv
      className={className}
    >
      <Normaltekst>
        {t('buc:form-noInstitutionYet')}
      </Normaltekst>
    </InstitutionListDiv>
  ) : (
    <>
      {Object.keys(institutionList).map(landkode => {
        const country = CountryData.getCountryInstance(locale).findByValue(landkode)
        return (
          <InstitutionListDiv
            className={className}
            key={landkode}
          >
            {type === 'joined' && (
              <InstitutionDiv>
                {flag && (
                  <Flag
                    className='mr-2'
                    label={country ? country.label : landkode}
                    country={landkode}
                    size='S'
                    type={flagType}
                  />
                )}
                <InstitutionText>
                  {institutionList[landkode].map((institutionId: string) => getLabel(landkode, institutionId)).join(', ')}
                </InstitutionText>
              </InstitutionDiv>
            )}
            {type === 'separated' && institutionList[landkode].map((institutionId : string) => (
              <InstitutionDiv key={institutionId}>
                {flag && (
                  <Flag
                    label={country ? country.label : landkode}
                    country={landkode}
                    size='S'
                    type={flagType}
                  />
                )}
                <InstitutionText>
                  {getLabel(landkode, institutionId)}
                </InstitutionText>
              </InstitutionDiv>
            ))}
          </InstitutionListDiv>
        )
      })}
    </>
  )
}

InstitutionList.propTypes = {
  className: PT.string,
  flag: PT.bool,
  flagType: PT.oneOf(['original', 'circle']),
  institutions: InstitutionsPropType.isRequired,
  locale: AllowedLocaleStringPropType.isRequired,
  type: PT.string
}

export default InstitutionList
