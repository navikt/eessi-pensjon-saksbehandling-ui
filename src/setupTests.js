import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// import { expect } from 'chai'
import cheerio from 'cheerio'

Enzyme.configure({ adapter: new Adapter() })

global.shallow = shallow
global.render = render
global.mount = mount

global.cheerio = cheerio
