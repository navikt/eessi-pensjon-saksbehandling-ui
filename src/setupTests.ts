import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { act } from 'react-dom/test-utils'
Enzyme.configure({ adapter: new Adapter() });

(global as any).shallow = shallow;
(global as any).render = render;
(global as any).mount = mount;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()
window.location.reload = jest.fn()
window.fetch = require('jest-fetch-mock')
Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})
