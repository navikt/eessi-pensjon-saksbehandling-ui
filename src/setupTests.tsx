
import { render, fireEvent, screen, getNodeText, cleanup } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { useDispatch, useSelector } from 'react-redux'
import 'jest-styled-components'
import '@testing-library/jest-dom'
import { TextEncoder, TextDecoder } from 'util';
(global as any).TextEncoder = TextEncoder;
(global as any).TextDecoder = TextDecoder;

jest.mock('react-redux');

(global as any).screen = screen;
(global as any).render = render;
(global as any).getNodeText = getNodeText;
(global as any).fireEvent = fireEvent;
(global as any).cleanup = cleanup;
(global as any).act = act

HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()

class ResizeObserver {
  observe () {}
  unobserve () {}
  disconnect () {}
}
window.ResizeObserver = ResizeObserver

const crypto = require('crypto')

Object.defineProperty(global.self, 'crypto', {
  value: {
    getRandomValues: (arr: any) => crypto.randomBytes(arr.length)
  }
})

Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})

jest.mock('uuid', () => ({
  v4: () => 'mock-uuid'
}))

jest.mock('i18next', () => {
  const use = jest.fn()
  const init = jest.fn()
  const loadLanguages = jest.fn()
  const t = jest.fn()
  const result = {
    use,
    init,
    loadLanguages,
    t
  }
  use.mockImplementation(() => result)
  return result
})

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock('react-pdf', () => {
  return {
    pdfjs: { GlobalWorkerOptions: { workerSrc: '' } },
    Document: (props: any) => {
      setTimeout(() => {
        props.onLoadSuccess({ numPages: 5 })
      }, 500)
      return (
        <div className='mock-pdfdocument'>
          {props.children}
          </div>
      )
    },
    Page: (props: any) => {
      return (
        <div className='mock-pdfpage'>
          {'Page: '}{props.pageNumber}
      </div>
    )
    }
  }
})

export const stageSelector = (defaultSelector: any, params: any) => {
  (useDispatch as jest.Mock).mockImplementation(() => jest.fn());
  (useSelector as jest.Mock).mockImplementation(() => ({
    ...defaultSelector,
    ...params
  }))
}
