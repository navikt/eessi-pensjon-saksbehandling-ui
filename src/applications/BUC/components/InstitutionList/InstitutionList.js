import React from 'react'
import PT from 'prop-types'
import _ from 'lodash'
import classNames from 'classnames'
import Flag from 'components/ui/Flag/Flag'
import CountryData from 'components/ui/CountryData/CountryData'
import { Element, Normaltekst } from 'components/ui/Nav'

import './InstitutionList.css'

const InstitutionList = (props) => {
  const { className, institutions, institutionNames, t, type, locale } = props

  let institutionList = {}
  if (institutions) {
    institutions.forEach(item => {
      if (institutionList.hasOwnProperty(item.country)) {
        institutionList[item.country].push(item.institution)
      } else {
        institutionList[item.country] = [item.institution]
      }
    })
  }

  if (_.isEmpty(institutionList)) {
    return <Normaltekst>{t('buc:form-noInstitutionYet')}</Normaltekst>
  }

  return Object.keys(institutionList).map(landkode => {
    const country = CountryData.findByValue(locale, landkode)
    return <div
      id='a-buc-c-institutionlist-id'
      className={classNames('a-buc-c-institutionlist', className)}
      key={landkode}>
      {type === 'joined' ? <div className='a-buc-c-institution'>
        <Flag label={country.label} country={landkode} size='M' />
        <Element className='pr-2 pl-2'>{country.label}: </Element>
        <Normaltekst>{institutionList[landkode].map(institutionId => {
          return institutionNames && institutionNames.hasOwnProperty(landkode + ':' + institutionId)
            ? institutionNames[landkode + ':' + institutionId] : institutionId
        }).join(', ')}</Normaltekst>
      </div> : null}
      {type === 'separated' ? institutionList[landkode].map(institutionId => {
        const label = institutionNames && institutionNames.hasOwnProperty(landkode + ':' + institutionId)
          ? institutionNames[landkode + ':' + institutionId] : institutionId
        return <div className='a-buc-c-institution' key={institutionId}>
          <Flag label={country.label} country={landkode} size='M' />
          <Element className='pr-2 pl-2'>{country.label}: </Element>
          <Normaltekst>{label}</Normaltekst>
        </div>
      }) : null}
    </div>
  })
}

InstitutionList.propTypes = {
  className: PT.string,
  t: PT.func.isRequired,
  type: PT.string.isRequired,
  institutions: PT.array.isRequired,
  institutionNames: PT.object,
  locale: PT.string.isRequired
}

export default InstitutionList
