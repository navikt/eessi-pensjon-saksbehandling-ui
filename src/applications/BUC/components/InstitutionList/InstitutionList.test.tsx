import { mount, ReactWrapper } from 'enzyme'
import { stageSelector } from 'setupTests'
import InstitutionList, { InstitutionListProps } from './InstitutionList'

// InstitutionNames
const defaultSelector = {
  'NO:Mock1': 'Mock 1 institution',
  'NO:Mock2': 'Mock 2 institution'
}

describe('applications/BUC/components/InstitutionList/InstitutionList', () => {
  let wrapper: ReactWrapper
  const initialMockProps: InstitutionListProps = {
    institutions: [{
      country: 'NO',
      institution: 'Mock1'
    }, {
      country: 'NO',
      institution: 'Mock2'
    }],
    locale: 'nb'
  } as InstitutionListProps

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<InstitutionList {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: Has proper HTML structure with joined type', () => {
    expect(wrapper.find('[data-test-id=\'a-buc-c-institutionlist__div-id\']').hostNodes()).toHaveLength(1)
    expect(wrapper.find('[data-test-id=\'a-buc-c-institutionlist__div-id\']').hostNodes().render().text()).toEqual('Mock1, Mock2')
  })

  it('Render: Has proper HTML structure with separated type', () => {
    wrapper = mount(<InstitutionList {...initialMockProps} type='separated' />)
    expect(wrapper.find('[data-test-id=\'a-buc-c-institutionlist__div-id\']').hostNodes()).toHaveLength(2)
    expect(wrapper.find('[data-test-id=\'a-buc-c-institutionlist__div-id\']').hostNodes().first().render().text()).toEqual('Mock1')
    expect(wrapper.find('[data-test-id=\'a-buc-c-institutionlist__div-id\']').hostNodes().last().render().text()).toEqual('Mock2')
  })
})
