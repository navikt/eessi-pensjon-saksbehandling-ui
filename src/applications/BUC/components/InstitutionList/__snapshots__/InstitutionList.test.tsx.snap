// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/InstitutionList/InstitutionList Render: match snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:not(.noMargin) {
  margin-bottom: 0.35rem;
}

ui-fss.g1[id="c2"] {
  content: "c3,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g2[id="c0"] {
  content: "c1,";
}

.c5 {
  margin-left: 0.5rem;
}

ui-fss.g3[id="c4"] {
  content: "c5,";
}

<InstitutionList
  institutions={
    Array [
      Object {
        "country": "NO",
        "institution": "Mock1",
      },
      Object {
        "country": "NO",
        "institution": "Mock2",
      },
    ]
  }
  locale="nb"
>
  <styled.div
    key="NO"
  >
    <div
      className="c0 c1"
    >
      <styled.div
        data-test-id="a-buc-c-institutionlist__div-id"
      >
        <div
          className="c2 c3"
          data-test-id="a-buc-c-institutionlist__div-id"
        >
          <Td
            country="NO"
            label="Norge"
            size="S"
            type="circle"
          >
            <Component
              highContrast={false}
            >
              <Pe
                theme={
                  Object {
                    "alternative-background-color": "#FFFFFF",
                    "alternative-hover-color": "#cce1f3",
                    "black": "#000000",
                    "border-radius-base": "4px",
                    "border-radius-large": "6px",
                    "border-radius-small": "2px",
                    "container-desktop": "(940px + @grid-gutter-width)",
                    "container-large-desktop": "(1140px + @grid-gutter-width)",
                    "container-lg": "(1140px + @grid-gutter-width)",
                    "container-md": "(940px + @grid-gutter-width)",
                    "container-sm": "(720px + @grid-gutter-width)",
                    "container-tablet": "(720px + @grid-gutter-width)",
                    "fokusFarge": "#19548a",
                    "grayBackground": "#e9e7e7",
                    "grayIcon": "#b7b1a9",
                    "grayInactive": "#78706a",
                    "grayModia": "#333333",
                    "grid-columns": "12",
                    "grid-float-breakpoint": "768px",
                    "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
                    "grid-gutter-width": "16px",
                    "inverted-font-color": "#FFFFFF",
                    "line-height-large": "1.3333333",
                    "line-height-small": "1.5",
                    "main-active-color": "#005b82",
                    "main-background-color": "#f1f1f1",
                    "main-border-color": "#78706a",
                    "main-border-radius": "4px",
                    "main-border-width": "1px",
                    "main-disabled-color": "#b7b1a9",
                    "main-error-color": "#ba3a26",
                    "main-focus-color": "#19548a",
                    "main-font-color": "#3e3832",
                    "main-hover-color": "#0074df",
                    "main-interactive-color": "#0067c5",
                    "main-success-color": "#06893a",
                    "main-warning-color": "#ff9100",
                    "navBakgrunn": "#ffffff",
                    "navBla": "#0067c5",
                    "navBlaDarken20": "#32414f",
                    "navBlaDarken40": "#254b6d",
                    "navBlaDarken60": "#19548a",
                    "navBlaDarken80": "#0c5ea8",
                    "navBlaLighten20": "#cce1f3",
                    "navBlaLighten40": "#99c2e8",
                    "navBlaLighten60": "#66a4dc",
                    "navBlaLighten80": "#3385d1",
                    "navDypBla": "#005b82",
                    "navDypBlaDarken20": "#323f42",
                    "navDypBlaDarken40": "#254652",
                    "navDypBlaDarken60": "#194d62",
                    "navDypBlaDarken80": "#0c5472",
                    "navDypBlaLighten20": "#ccdee6",
                    "navDypBlaLighten40": "#99bdcd",
                    "navDypBlaLighten60": "#669db4",
                    "navDypBlaLighten80": "#337c9b",
                    "navGra20": "#c6c2bf",
                    "navGra40": "#b7b1a9",
                    "navGra60": "#78706a",
                    "navGra80": "#59514b",
                    "navGraBakgrunn": "#f1f1f1",
                    "navGronn": "#06893a",
                    "navGronnDarken20": "#334834",
                    "navGronnDarken40": "#285835",
                    "navGronnDarken60": "#1c6937",
                    "navGronnDarken80": "#117938",
                    "navGronnLighten20": "#cde7d8",
                    "navGronnLighten40": "#9bd0b0",
                    "navGronnLighten60": "#6ab889",
                    "navGronnLighten80": "#38a161",
                    "navLilla": "#634689",
                    "navLillaDarken20": "#453b43",
                    "navLillaDarken40": "#4d3e55",
                    "navLillaDarken60": "#544066",
                    "navLillaDarken80": "#5c4378",
                    "navLillaLighten20": "#e0dae7",
                    "navLillaLighten40": "#c1b5d0",
                    "navLillaLighten60": "#a190b8",
                    "navLillaLighten80": "#826ba1",
                    "navLimeGronn": "#a2ad00",
                    "navLimeGronnDarken20": "#524f28",
                    "navLimeGronnDarken40": "#66671e",
                    "navLimeGronnDarken60": "#7a7e14",
                    "navLimeGronnDarken80": "#8e960a",
                    "navLimeGronnLighten20": "#ecefcc",
                    "navLimeGronnLighten40": "#dade99",
                    "navLimeGronnLighten60": "#c7ce66",
                    "navLimeGronnLighten80": "#b5bd33",
                    "navLysBla": "#66cbec",
                    "navLysBlaDarken20": "#465557",
                    "navLysBlaDarken40": "#4e737c",
                    "navLysBlaDarken60": "#5690a2",
                    "navLysBlaDarken80": "#5eaec7",
                    "navLysBlaLighten20": "#e0f5fb",
                    "navLysBlaLighten40": "#c2eaf7",
                    "navLysBlaLighten60": "#a3e0f4",
                    "navLysBlaLighten80": "#85d5f0",
                    "navLysGra": "#e9e7e7",
                    "navMorkGra": "#3e3832",
                    "navOransje": "#ff9100",
                    "navOransjeDarken20": "#654a28",
                    "navOransjeDarken40": "#8b5c1e",
                    "navOransjeDarken60": "#b26d14",
                    "navOransjeDarken80": "#d87f0a",
                    "navOransjeLighten20": "#ffe9cc",
                    "navOransjeLighten40": "#ffd399",
                    "navOransjeLighten60": "#ffbd66",
                    "navOransjeLighten80": "#ffa733",
                    "navRod": "#c30000",
                    "navRodDarken20": "#592d28",
                    "navRodDarken40": "#73221e",
                    "navRodDarken60": "#8e1614",
                    "navRodDarken80": "#a80b0a",
                    "navRodLighten20": "#f3cccc",
                    "navRodLighten40": "#e79999",
                    "navRodLighten60": "#db6666",
                    "navRodLighten80": "#cf3333",
                    "orangeFocus": "#ffbd66",
                    "orangeFocusDarken20": "#65533c",
                    "orangeFocusDarken40": "#8b6d47",
                    "orangeFocusDarken60": "#b28851",
                    "orangeFocusDarken80": "#d8a25c",
                    "orangeFocusLighten20": "#fff2e0",
                    "orangeFocusLighten40": "#ffe5c2",
                    "orangeFocusLighten60": "#ffd7a3",
                    "orangeFocusLighten80": "#ffca85",
                    "padding-base-horizontal": "12px",
                    "padding-base-vertical": "6px",
                    "padding-large-horizontal": "16px",
                    "padding-large-vertical": "10px",
                    "padding-small-horizontal": "10px",
                    "padding-small-vertical": "5px",
                    "padding-xs-horizontal": "5px",
                    "padding-xs-vertical": "1px",
                    "pinkErrorBg": "#f3e3e3",
                    "redError": "#ba3a26",
                    "redErrorDarken20": "#573830",
                    "redErrorDarken40": "#70392d",
                    "redErrorDarken60": "#88392b",
                    "redErrorDarken80": "#a13a28",
                    "redErrorLighten20": "#f1d8d4",
                    "redErrorLighten40": "#e3b0a8",
                    "redErrorLighten60": "#d6897d",
                    "redErrorLighten80": "#c86151",
                    "screen-desktop": "992px",
                    "screen-lg": "1200px",
                    "screen-lg-desktop": "1200px",
                    "screen-lg-min": "1200px",
                    "screen-md": "992px",
                    "screen-md-max": "(@screen-lg-min - 1)",
                    "screen-md-min": "992px",
                    "screen-sm": "768px",
                    "screen-sm-max": "(@screen-md-min - 1)",
                    "screen-sm-min": "768px",
                    "screen-tablet": "768px",
                    "screen-xs-max": "(@screen-sm-min - 1)",
                    "type": "theme",
                    "white": "#FFFFFF",
                  }
                }
              >
                <ForwardRef(Ad)
                  overlay={
                    <span>
                      Norge
                    </span>
                  }
                  placement="top"
                  trigger={
                    Array [
                      "hover",
                    ]
                  }
                >
                  <t
                    action={
                      Array [
                        "hover",
                      ]
                    }
                    afterPopupVisibleChange={[Function]}
                    autoDestroy={false}
                    blurDelay={0.15}
                    builtinPlacements={
                      Object {
                        "bottom": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tc",
                            "bc",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "bottomLeft": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tl",
                            "bl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "bottomRight": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tr",
                            "br",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "left": Object {
                          "offset": Array [
                            -4,
                            0,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "cr",
                            "cl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "leftBottom": Object {
                          "offset": Array [
                            -4,
                            0,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "br",
                            "bl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "leftTop": Object {
                          "offset": Array [
                            -4,
                            0,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tr",
                            "tl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "right": Object {
                          "offset": Array [
                            4,
                            0,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "cl",
                            "cr",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "rightBottom": Object {
                          "offset": Array [
                            4,
                            0,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bl",
                            "br",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "rightTop": Object {
                          "offset": Array [
                            4,
                            0,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tl",
                            "tr",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "top": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bc",
                            "tc",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topLeft": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bl",
                            "tl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topRight": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "br",
                            "tr",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                      }
                    }
                    defaultPopupVisible={false}
                    destroyPopupOnHide={false}
                    focusDelay={0}
                    getDocument={[Function]}
                    getPopupClassNameFromAlign={[Function]}
                    hideAction={Array []}
                    mask={false}
                    maskClosable={true}
                    mouseEnterDelay={0}
                    mouseLeaveDelay={0.1}
                    onPopupAlign={[Function]}
                    onPopupVisibleChange={[Function]}
                    overlay={
                      <span>
                        Norge
                      </span>
                    }
                    popup={[Function]}
                    popupAlign={Object {}}
                    popupClassName=""
                    popupPlacement="top"
                    popupStyle={Object {}}
                    prefixCls="rc-tooltip"
                    showAction={Array []}
                  >
                    <styled.div
                      className="size-S type-circle animate wave"
                      key="trigger"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={Object {}}
                    >
                      <div
                        className="hxgtkf size-S type-circle animate wave"
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={Object {}}
                      >
                        <styled.div
                          className="size-S type-circle"
                        >
                          <div
                            className="JXIdR size-S type-circle"
                          >
                            <img
                              alt="Norge"
                              src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1100 800'%3e %3crect width='1100' height='800' fill='%23ef2b2d'/%3e %3crect width='200' height='800' x='300' fill='%23fff'/%3e %3crect width='1100' height='200' y='300' fill='%23fff'/%3e %3crect width='100' height='800' x='350' fill='%23002868'/%3e %3crect width='1100' height='100' y='350' fill='%23002868'/%3e %3c/svg%3e"
                              style={Object {}}
                            />
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </t>
                </ForwardRef(Ad)>
              </Pe>
            </Component>
          </Td>
          <Styled(Normaltekst)>
            <Normaltekst
              className="c4 c5"
            >
              <TypografiBase
                className="c4 c5"
                type="normaltekst"
              >
                <p
                  className="typo-normal c4 c5"
                >
                  Mock1, Mock2
                </p>
              </TypografiBase>
            </Normaltekst>
          </Styled(Normaltekst)>
        </div>
      </styled.div>
    </div>
  </styled.div>
</InstitutionList>
`;
